src/spatialdata/_core/query/spatial_query.py

```diff
@@ -74,38 +75,65 @@ def _get_bounding_box_corners_in_intrinsic_coordinates(
 
     The axes of the intrinsic coordinate system.
     """
-    from spatialdata.transformations import get_transformation
-
     min_coordinate = _parse_list_into_array(min_coordinate)
     max_coordinate = _parse_list_into_array(max_coordinate)
-    # get the transformation from the element's intrinsic coordinate system
-    # to the query coordinate space
-    transform_to_query_space = get_transformation(element, to_coordinate_system=target_coordinate_system)
+
+    # compute the output axes of the transformation, remove c from input and output axes, return the matrix without c
+    # and then build an affine transformation from that
     m_without_c, input_axes_without_c, output_axes_without_c = _get_axes_of_tranformation(
         element, target_coordinate_system
     )
+    spatial_transform = Affine(m_without_c, input_axes=input_axes_without_c, output_axes=output_axes_without_c)
+
+    # we identified 5 cases (see the responsible function for details), cases 1 and 5 correspond to invertible
+    # transformations; we focus on them
+    if isinstance(element, (DataArray, DataTree)):
+        m_without_c_linear = m_without_c[:-1, :-1]
+        case = _get_case_of_bounding_box_query(m_without_c_linear, input_axes_without_c, output_axes_without_c)
+        assert case in [1, 5]
+
+    # adjust the bounding box to the real axes, dropping or adding eventually mismatching axes; the order of the axes is
+    # not adjusted
     axes, min_coordinate, max_coordinate = _adjust_bounding_box_to_real_axes(
         axes, min_coordinate, max_coordinate, output_axes_without_c
     )
+    assert set(axes) == set(output_axes_without_c)
+
+    # in the case of non-raster data, we need to swap the axes
+    if not isinstance(element, (DataArray, DataTree)):
+        input_axes_without_c, axes = axes, input_axes_without_c
```

Comment:
are you sure? I think before we were using this variable after calling inverse, and now we first construct the affine and then call inverse. So I don't think it's a case "raster" vs "non-raster", and the same code should work for both.

Reply:
yeah I am not 100% sure but if I don't swap the axes, this test fails
```
FAILED tests/core/query/test_spatial_query.py::test_query_spatial_data - AssertionError: DataFrame are different
```

Reply:
I think it has to do more with the fact that I have to unify the variables, and that I am now using the same variables for raster data, also for non-raster data, you can see how this function is used in the non-raster case before this PR. There maybe be a better way that doesn't require this but then the input/output axes of the transformation in the non raster case would have to be changed.

Reply:
Thanks for the info. I'll also have a look at the code.

Reply:
Fixed, I explained this in the conversation.